bull-bean-transformer/src/main/java/com/hotels/beans/transformer/TransformerImpl.java

```diff
@@ -90,6 +97,34 @@
         }
     }
 
+    /**
+     * Inject all properties value from an object to a new one.
+     * @param sourceObj the source object
+     * @param targetClass the destination object class
+     * @param breadcrumb the full path of the current field starting from his ancestor
+     * @param <T> the Source object type
+     * @param <K> the target object type
+     * @return a copy of the source object into the destination object
+     */
+    private <T, K> K injectValues(final T sourceObj, final Class<? extends K> targetClass, final String breadcrumb) {
+        final K k;
+        final ClassType classType = classUtils.getClassType(targetClass);
+        if (classType.is(MUTABLE)) {
+            try {
+                k = classUtils.getNoArgsConstructor(targetClass).get();
+                injectAllFields(sourceObj, k, breadcrumb);
+            } catch (Exception e) {
+                throw new InvalidBeanException(e.getMessage(), e);
+            }
+        } else {
+            k = injectValues(sourceObj, targetClass, classUtils.getAllArgsConstructor(targetClass), breadcrumb, false);
+            if (classType.is(MIXED)) {
+                injectNotFinalFields(sourceObj, k, breadcrumb);
```

Comment:
Should we be handling an exception here also, like in the above `if` block?

Reply:
we can avoid handling an exception as it's already managed inside the method and the one throws needs to be raised.
Different is for the previous block, where there is the default constructor invocation (that is a `Supplier`) and the exception thrown is not managed.